tests/cli/test_cmds_local_run.py

```diff
@@ -1959,3 +1960,352 @@ def test_missing_volumes_skipped(mock_exists):
         )
         args, kwargs = mock_run_docker_container.call_args
         assert kwargs["volumes"] == []
+
+
+@mock.patch("paasta_tools.cli.cmds.local_run.pick_random_port", autospec=True)
+@mock.patch("paasta_tools.cli.cmds.local_run.get_docker_run_cmd", autospec=True)
+@mock.patch("paasta_tools.cli.cmds.local_run.execlpe", autospec=True)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run._run",
+    autospec=True,
+    return_value=(0, "fake _run output"),
+)
+@mock.patch("paasta_tools.cli.cmds.local_run.get_container_id", autospec=True)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run.get_healthcheck_for_instance",
+    autospec=True,
+    return_value=("fake_healthcheck_mode", "fake_healthcheck_uri"),
+)
+@mock.patch("paasta_tools.cli.cmds.local_run.decrypt_secret_volumes", autospec=True)
+@mock.patch("paasta_tools.cli.cmds.local_run.shutil", autospec=True)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run.open",
+    new_callable=mock.mock_open(),
+    autospec=None,
+)
+@mock.patch("os.makedirs", autospec=True)
```

Comment:
imo, it's a little preferable to use the context manager form of mocking - but we take the decorator approach in enough places already that this is probably fine :)

Reply:
Yeah, this is unfortunate, especially since I care zero about so many of those mocks for the scope of what I actually care to test in my test. It'd be nice if we had an abstraction like https://sourcegraph.yelpcorp.com/services/actionable_alerting_service/-/blob/tests/service_test.py?L23 to DRY this code up a bit, but I am going to respectfully decline / defer that here :)